libraries/ms-drivers/src/x86/mcp3427_adc.c

```diff
@@ -64,8 +106,18 @@ static void prv_raise_ready(SoftTimerId timer_id, void *context) {
 
 static void prv_channel_ready(struct Fsm *fsm, const Event *e, void *context) {
   Mcp3427Storage *storage = (Mcp3427Storage *)context;
+#ifdef MPXE
+  s_store.readings[storage->current_channel] = FIXED_RESULT;
```

Comment:
Setting the store here isn't necessary. Since you are updating the store from the mpxe harness, you don't need to change the values going into the harness from the c side, they only need to go python -> c.  After you get rid of this line, the smoke tests might fail because they are expecting the fixed value. In your integration tests on the python side, set the mcp3427 value to the expected (20) and then the smoke test will read this value. Should be a minor change 🙂